main.go

```diff
@@ -1,159 +1,191 @@
-
+	"github.com/gorilla/sessions"
 	"github.com/joeshaw/envdecode"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 )
 
-type Config struct {
-	ClientId      string `env:"CLIENT_ID,required"`      // Google Client ID
-	ClientSecret  string `env:"CLIENT_SECRET,required"`  // Google Client Secret
-	SessionSecret string `env:"SESSION_SECRET,required"` // Random session encruption token
-	DNSName       string `env:"DNS_NAME,required"`       // Public facing DNS Hostname
-
-	SessionDBPath string `env:"SESSION_DB_PATH,default=./sessions.db"` // Path to session database, including db name
-	CookieMaxAge  int    `env:"COOKIE_MAX_AGE,default=1440"`           // Cookie MaxAge, Defaults to 1 day
-	CookieName    string `env:"COOKIE_NAME,default=sproxy_session"`    // The name of the cookie
-
-	ProxyURL string `env:"PROXY_URL,default=http://localhost:8000/"` // URL to Proxy to
-
-	CallBackPath string `env:"CALLBACK_PATH,default=/auth/callback/google"` // Callback URL
-	AuthPath     string `env:"AUTH_PATH,default=/auth/google"`              // Auth Path
-
-	HealthCheckPath string `env:"HEALTH_CHECK_PATH,default=/en-US/static/html/credit.html"` // Health Check path in splunk, this path is proxied w/o auth. The default is a static file served by the splunk web server
+type config struct {
+	clientID              string   `env:"CLIENT_ID,required"`                                       // Google Client ID
+	clientSecret          string   `env:"CLIENT_SECRET,required"`                                   // Google Client Secret
+	sessionSecret         string   `env:"SESSION_SECRET,required"`                                  // Random session auth key
+	sessionEncrypttionKey string   `env:"SESSION_ENCRYPTION_KEY,required"`                          // Random session encryption key
+	dnsName               string   `env:"DNS_NAME,required"`                                        // Public facing DNS Hostname
+	cookieMaxAge          int      `env:"COOKIE_MAX_AGE,default=1440"`                              // Cookie MaxAge, Defaults to 1 day
+	cookieName            string   `env:"COOKIE_NAME,default=sproxy_session"`                       // The name of the cookie
+	proxyURL              *url.URL `env:"PROXY_URL,default=http://localhost:8000/"`                 // URL to Proxy to
+	callbackPath          string   `env:"CALLBACK_PATH,default=/auth/callback/google"`              // Callback URL
+	healthCheckPath       string   `env:"HEALTH_CHECK_PATH,default=/en-US/static/html/credit.html"` // Health Check path in splunk, this path is proxied w/o auth. The default is a static file served by the splunk web server
+	emailSuffix           string   `env:"EMAIL_SUFFIX,default=@heroku.com"`                         // Required email suffix. Emails w/o this suffix will not be let in
+	stateToken            string   `env:"STATE_TOKEN,required"`                                     // Token used when communicating with Google Oauth2 provider
```

Comment:
It could be, but then we would have to track those in a database of some sort (psql / redis / etc). If we use a private token then at least we have a little protection.